### PR TITLE
ARROW-7468: [Python] fix typos

### PR DESCRIPTION
**Pull request by kiszk**
This PR fixes typos in files under `python` directory.

### CODE REVIEW

File: python/pyarrow/tests/test_parquet.py [hunk 1]
```diff
@@ -2803,24 +2803,24 @@ def _make_dataset_for_pickling(tempdir, N=100):
     pytest.param(pytest.importorskip('cloudpickle'), id='cloudpickle')
 ])
 def test_pickle_dataset(tempdir, datadir, pickler):
-    def is_pickleable(obj):
+    def is_picklable(obj):
```
Comment (kou) [inline python/pyarrow/tests/test_parquet.py]: I think that "pickle"-able is valid.

Comment (kiszk) [reply python/pyarrow/tests/test_parquet.py]: I see. I am neutral on this. I just followed [this API](https://jsonpickle.github.io/api.html#jsonpickle.util.is_picklable).   
I have just reverted these changes.

Comment (kou) [reply python/pyarrow/tests/test_parquet.py]: OK. I'll merge this pull request without these changes for now.
If some Python experts say "picklable" is better, we will use "picklable".

**Review by kou — APPROVED** [review]
+1

File: python/pyarrow/tests/test_parquet.py (hunk 1 above)
Comment (wesm) [reply python/pyarrow/tests/test_parquet.py]: The Python documentation uses "picklable" https://docs.python.org/3/library/pickle.html

Comment (kou) [reply python/pyarrow/tests/test_parquet.py]: Oh... Sorry for my wrong comment...

**Pull request closed without merging**